test/distributed/test_inductor_collectives.py

```diff
@@ -121,6 +121,42 @@ def compile(func, example_inputs):
             inductor_out = compiled_matmul_cat_col(*inputs)
             self.assertTrue(same(eager_out, inductor_out, tol=0.001))
 
+    @unittest.skipIf(not has_triton(), "Inductor+gpu needs triton and recent GPU arch")
+    @skip_if_lt_x_gpu(2)
+    # TODO: somehow inductor bg compile threads are causing hangs at exit with distributed work dtor
+    @patch.object(torch._inductor.config, "compile_threads", 1)
+    @patch.object(torch._inductor.config, "enable_comm_fusion_pass", True)
+    def test_comm_fusion_inductor(self):
+        """Test that inductor fx pass fuses comm_fusion patterns."""
+
+        def allreduce_split(a, idx, split_size, *, tag, ranks, group_size):
+            ar = torch.ops.c10d_functional.all_reduce(a, "sum", tag, ranks, group_size)
```

Comment:
Maybe worth having some tests for when we don't match successfully?

(1) more than one output from the `aten.split()` calls is consumed by a downstream op
(2) split index is not equal to current local rank